README.md

```diff
@@ -501,8 +501,66 @@ container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity
     'biocontainers/fastqc:0.11.9--0' }"
 ```
 
+For IRIDA Next Nextflow pipelines running in Azure and using Docker, the default container registry should be set to the publicly available `quay.io` registry in the `nextflow.config` file using the `docker.registry` parameter.
+
 For more information, see the [Nextflow containers][] documentation and the [nf-core modules software requirements][] guide.
 
+### 5.2.2 Configuring Module Software with Private Container Registries
+
+In some cases, you may want to use a private container registry instead of, or in addition to, a public container registry to define software requirements for Nextflow pipeline modules. For this purpose, we introduce the `--override_default_container_registry` parameter. This boolean parameter allows the Nextflow pipeline to override a default container registry with an alternative registry. By default, this parameter is set to `false`, and the default container registry, specified in the `nextflow.config` file, is used.
+
+To specify an alternative registry for a module, the `container` directive should incorporate a conditional check using the `params.override_default_container_registry` within a nested ternary operator in its closure.
+
+Below is an example of a `container` directive specifying the software requirements for a module in an IRIDA Next Nextflow pipeline:
+
+```
+container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    'https://depot.galaxyproject.org/singularity/fastqc:0.11.9--0' :
+    (params.override_default_container_registry ? 'acrgspimages.io/biocontainers/fastqc:0.11.9--0' : 'biocontainers/fastqc:0.11.9--0') }"
```

Comment:
What if in addition to the boolean `--override_default_container_registry` there is a parameter (`--alt_registry`) to prefix the alternative repositories as to not write specific registeries into the process? I'm assuming they'd always be the same when `--override_default_container_regsitry` is `true`.
```(params.override_default_container_registry ? '${alt_registry}/biocontainers/fastqc:0.11.9--0' : 'biocontainers/fastqc:0.11.9--0') }```

Reply:
So initially I thought the same thing and tried to set an `--alternative_registry` but I could not get this to work. I figured it had something to do with complications from scoping and evaluation order when accessing global parameters in quotes (' ') when using a nested ternary operator.

BUT...I realized Groovy can handle nested access to global parameters IF you use double quotes! (" ") 

SO THIS WORKS!

`container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/fastp:0.23.4--h5f740d0_0' :
        (params.override_default_container_registry ? "${params.alt_reg}/biocontainers/fastp:v0.20.1_cv1" : 
          'biocontainers/fastp:0.23.4--h5f740d0_0') }"
 `